src/rpc/core_rpc_server.cpp

```diff
@@ -159,6 +159,14 @@ namespace cryptonote
       const std::vector<std::string> ssl_allowed_fingerprint_strings = command_line::get_arg(vm, arg_rpc_ssl_allowed_fingerprints);
       std::vector<std::vector<uint8_t>> ssl_allowed_fingerprints{ ssl_allowed_fingerprint_strings.size() };
       std::transform(ssl_allowed_fingerprint_strings.begin(), ssl_allowed_fingerprint_strings.end(), ssl_allowed_fingerprints.begin(), epee::from_hex::vector);
+      for (const auto &fpr: ssl_allowed_fingerprints)
+      {
+        if (fpr.size() != 32)
```

Comment:
For added clarity we could use the `SHA256_DIGEST_LENGTH` constant here, as well as the other locations where 32 is hard-coded.

Reply:
I guess, but it's SHA256, it's already in the name (and it makes the error messages have to be constructed at runtime which is a pain) :)

Reply:
Sure, the fact that it's SHA256 does make it sort of obvious why it's 32 bytes. Personally I like to be verbose in these situations, if only because it makes grepping easier.

Messages don't necessarily have to be constructed at runtime though. With some preprocessor magic you can turn this numeric constant into a string literal, which you could then use. Something like this:

```
#define STR_HELPER(x) #x
#define STR(x) STR_HELPER(x)
```

You could then make your error message like this, which makes this a compile-time string literal:
`MERROR("SHA-256 fingerprint should be " STR(SHA256_DIGEST_LENGTH) " bytes long");`

I do wonder whether that's not taking it too far, though. The benefit I see of this is that if we would want to change the hashing algorithm for fingerprints yet again (say we want to go to sha384) it makes it easier to find all the places to change.

Reply:
OK, constructed at preprocess time which is a pain :)

Reply:
I'll change it.